apps/dolly-frontend/src/main/js/src/components/fagsystem/pdlf/visning/partials/DoedfoedtBarn.tsx

```diff
@@ -3,37 +3,99 @@ import { DollyFieldArray } from '@/components/ui/form/fieldArray/DollyFieldArray
 import { ErrorBoundary } from '@/components/ui/appError/ErrorBoundary'
 import { TitleValue } from '@/components/ui/titleValue/TitleValue'
 import { formatDate } from '@/utils/DataFormatter'
-import { DoedfoedtBarn } from '@/components/fagsystem/pdlf/PdlTypes'
+import { DoedfoedtBarnData, PersonData } from '@/components/fagsystem/pdlf/PdlTypes'
+import * as _ from 'lodash-es'
+import { initialDoedfoedtBarn } from '@/components/fagsystem/pdlf/form/initialValues'
+import { OpplysningSlettet } from '@/components/fagsystem/pdlf/visning/visningRedigerbar/OpplysningSlettet'
+import VisningRedigerbarConnector from '@/components/fagsystem/pdlf/visning/visningRedigerbar/VisningRedigerbarConnector'
+import React from 'react'
 
-type DoedfoedtBarnData = {
-	data: Array<DoedfoedtBarn>
+type DataListe = {
+	data: Array<DoedfoedtBarnData>
+	tmpPersoner?: Array<PersonData>
+	ident?: string
 }
 
-type VisningData = {
-	data: DoedfoedtBarn
+type Data = {
+	doedfoedtBarnData?: DoedfoedtBarnData
+	idx: number
+	data: Array<DoedfoedtBarnData>
+	tmpPersoner?: Array<PersonData>
+	ident?: string
 }
 
-export const Visning = ({ data }: VisningData) => (
-	<>
-		<ErrorBoundary>
-			<div className="person-visning_content">
-				<TitleValue title="Dødsdato" value={formatDate(data.dato)} />
-			</div>
-		</ErrorBoundary>
-	</>
-)
-
-export const DoedfoedtBarnVisning = ({ data }: DoedfoedtBarnData) => {
+const DoedfoedtBarnLes = ({ data, idx }: Data) => {
+	if (!data) {
+		return null
+	}
+	return (
+		<>
+			<ErrorBoundary>
+				<div className="person-visning_content" key={idx}>
+					<TitleValue title="Dødsdato" value={formatDate(data.dato)} />
+				</div>
+			</ErrorBoundary>
+		</>
+	)
+}
+
+export const DoedfoedtBarnVisning = ({
+	doedfoedtBarnData,
+	idx,
+	data,
+	tmpPersoner,
+	ident,
+}: Data) => {
+	const initDoedfoedtBarn = Object.assign(_.cloneDeep(initialDoedfoedtBarn), data[idx])
+	let initialValues = { doedfoedtBarn: initDoedfoedtBarn }
+
+	const redigertDoedfoedtBarnPdlf = _.get(tmpPersoner, `${ident}.person.doedfoedtBarn`)?.find(
+		(a: DoedfoedtBarnData) => a.id === doedfoedtBarnData.id
+	)
+
+	const slettetDoedfoedtBarnPdlf = tmpPersoner?.hasOwnProperty(ident) && !redigertDoedfoedtBarnPdlf
+	if (slettetDoedfoedtBarnPdlf) {
+		return <OpplysningSlettet />
+	}
+
+	const doedfoedtBarnValues = redigertDoedfoedtBarnPdlf
+		? redigertDoedfoedtBarnPdlf
+		: doedfoedtBarnData
+	let redigertDoedfoedtBarnValues = redigertDoedfoedtBarnPdlf
+		? {
+				doedfoedtBarn: Object.assign(_.cloneDeep(initialDoedfoedtBarn), redigertDoedfoedtBarnPdlf),
+		  }
```

Comment:
Denne kan vel byttes med `redigertDoedfoedtBarnPdlf && {.. `for bittelitt mindre kode 🤓

Reply:
Bra sett 🤓